CACHING.md

```diff
@@ -0,0 +1,351 @@
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: cache-${{ hashFiles('**/lockfiles') }}
+```
+
+### Using restore keys to download the closest matching cache
+
+If cache is not found matching the primary key, restore keys will be used to download the closest matching cache that was recently created. This way most of the dependencies can be downloaded from the restore key hence saving some build time.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: cache-npm-${{ hashFiles('**/lockfiles') }}
+      restore-keys: |
+        cache-npm-
+```
+
+The restore keys can be provided as a complete name, or a prefix, read more [here](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) on how a cache key is matched using restore keys.
+
+### Separate caches by Operating System
+
+In case of workflows with matrix running for multiple Operating Systems, the caches can be stored separately for each of them. This can be used in combination with hashfiles in case multiple caches are being generated per OS.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-cache
+```
+
+### Limiting cache to only the current workflow/attempt
+
+Caches scoped to the particular workflow run id or run attempt can be stored and referred by using the run id/attempt
+
+```yaml
+    key: cache-${{ github.run_id }}-${{ github.run_attempt }}
+```
+
+### Limiting cache for a particular commit
+
+For very short term or isolated use cases, where cache is supposed to be short lived, commit sha can be used.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: cache-${{ github.sha }}
+```
+
+### Using multiple factors while forming a key depening on the need
+
+Cache key can be formed by combination of more than one metadata, evaluated info.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
+
+The [GitHub Context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) can be used to create keys using the workflows metadata.
+
+## Restoring Cache
+
+### Know paths better
+
+While setting paths for caching dependencies it is important to give correct path depending on the image you are using and if you are running the action inside or outside the container. Below are some paths one should keep in mind while writing their workflow files.
+
+#### Ubuntu Paths
+
+```yaml
+process.env['RUNNER_TEMP']=/home/runner/work/_temp 
+process.cwd()=/home/runner/work/repo/repo
+Home directory = /home/runner
+${{ github.workspace }} = /home/runner/work/repo/repo
```

Comment:
Changed the order a bit based on priority of usage
```suggestion
Home directory (`~/`) = `/home/runner`
`${{ github.workspace }}` = `/home/runner/work/repo/repo`
`process.env['RUNNER_TEMP']`=`/home/runner/work/_temp` 
`process.cwd()` = `/home/runner/work/repo/repo`
```

Reply:
If okay, then change similar for other OSes as well

Reply:
The order makes sense, I'll change at all places.